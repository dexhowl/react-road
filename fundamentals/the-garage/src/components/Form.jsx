
export default function Form({children, formAction}) {


    return (
        <form action={formAction}>
            {children}
        </form>
    )
}