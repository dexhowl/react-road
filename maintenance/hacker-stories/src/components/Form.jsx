import Input from './Input'

export default function Form({userInput, formAction, onSearch}) {


    return (
        <form action={formAction}>
            <Input id={'search'} text={userInput} onSearch={onSearch}  isFocused>
                <strong>Search: </strong>
            </Input>
        </form>
    )
}