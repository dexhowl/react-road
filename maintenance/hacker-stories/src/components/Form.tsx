import * as React from 'react';

interface FormProps {
    children?: React.ReactNode;
    onSubmit?: React.FormEventHandler<HTMLFormElement>;
    
}

export default function Form({onSubmit, children}: FormProps) {
    return (
        <>
            <form action="" onSubmit={onSubmit}>
                {children}
            </form>
        </>
    )
}