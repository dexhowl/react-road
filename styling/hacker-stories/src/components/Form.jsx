import styled from 'styled-components'

export default function Form({children, formAction}) {

    const StyledForm = styled.form`
        padding: 10px 0 20px 0;
        display: flex;
        align-items: baseline;
  `;
  

    return (
        <StyledForm className="form" action={formAction}>
            {children}
        </StyledForm>
    )
}