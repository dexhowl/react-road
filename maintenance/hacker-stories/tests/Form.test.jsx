/* globals describe render screen afterEach vi it renderHook act*/

import Form from '../src/components/Form'

describe('Form',() => {
    it('Renders',() => {
        render(<Form/>);
        
    })
    it('Accepts props',() => {
        
    })
    
})