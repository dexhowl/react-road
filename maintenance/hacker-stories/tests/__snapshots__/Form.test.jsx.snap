// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Form > renders snapshot 1`] = `
<form
  action="javascript:throw new Error('A React form was unexpectedly submitted. If you called form.submit() manually, consider using form.requestSubmit() instead. If you\\'re trying to use event.stopPropagation() in a submit event handler, consider also calling event.preventDefault().')"
>
  <label
    for="search"
  >
    <strong>
      Search: 
    </strong>
  </label>
  <input
    id="search"
    type="text"
    value="Runescape"
  />
  <button
    type="submit"
  >
    Search
  </button>
</form>
`;
